user.go

```diff
@@ -53,11 +60,18 @@ func createUser(c tacitContext, db tacitDB) {
 
 	theUser := dbUser{Username: aUser.Username}
 
+	//Uses gorm RecordNotFound
+	if !db.recordNotFound() {
+		fmt.Printf("The username is already taken.")
```

Comment:
I don't think this works quite the way you think it does. Without db ever being constraied or called upon to make any explixit call to the db we can expect `db.Error` to always be `nil` here I think we would have to at least run this after the create. Also I am a bit suspicious that this double negative actually does not do the right thing. since `db.recordNotFound()` will return weather the `db.Error` is of type `RecordNotFound` that means if there is no error this will be false right out of the gate and will yield a false negative for us here. We should revisit this one!